sympy/solvers/ode.py

```diff
@@ -2410,6 +2410,20 @@ def checkodesol(ode, sol, func=None, order='auto', solve_for_func=True):
     x = func.args[0]
     while s:
         if testnum == 0:
+            ode_expr = ode.lhs - ode.rhs
+            temp = None
+            if sol.lhs == func and not sol.rhs.has(func):
+                temp = sol.rhs
+            elif sol.rhs == func and not sol.lhs.has(func):
+                temp = sol.lhs
+            if temp is not None:
+                for i in range(order,-1,-1):
+                    ode_expr = simplify(ode_expr.subs(func.diff(x, i), temp.diff(x, i)))
```

Comment:
I haven't tested but it's probably better to only call simplify once after all substitutions are done.